core/src/test/java/org/apache/iceberg/rest/responses/TestLoadTableResponse.java

```diff
@@ -0,0 +1,371 @@
+            TEST_SCHEMA, SPEC_5, SORT_ORDER_3, TEST_TABLE_LOCATION, TABLE_PROPS))
+        .withMetadataLocation(TEST_METADATA_LOCATION)
+        .build();
+
+    LoadTableResponse respV1 = LoadTableResponse.builder()
+        .withTableMetadata(metaV1)
+        .addAllConfig(CONFIG)
+        .build();
+
+    String fullJson = "{" +
+        "  \"metadataLocation\":\"s3://bucket/test/location/metadata/v1.metadata.json\"," +
+        "  \"metadata\":{" +
+        "    \"format-version\":1," +
+        "    \"table-uuid\":\"" + metaV1.uuid() + "\"," +
+        "    \"location\":\"s3://bucket/test/location\"," +
+        "    \"last-updated-ms\":" + metaV1.lastUpdatedMillis() + "," +
+        "    \"last-column-id\":3," +
+        "    \"schema\":{\"type\":\"struct\",\"schema-id\":0,\"fields\":[" +
+        "        {\"id\":1,\"name\":\"x\",\"required\":true,\"type\":\"long\"}," +
+        "        {\"id\":2,\"name\":\"y\",\"required\":true,\"type\":\"long\",\"doc\":\"comment\"}," +
+        "        {\"id\":3,\"name\":\"z\",\"required\":true,\"type\":\"long\"}" +
+        "       ]" +
+        "    }," +
+        "  \"current-schema-id\":0," +
+        "  \"schemas\":[" +
+        "    {\"type\":\"struct\",\"schema-id\":0,\"fields\":[" +
+        "      {\"id\":1,\"name\":\"x\",\"required\":true,\"type\":\"long\"}," +
+        "      {\"id\":2,\"name\":\"y\",\"required\":true,\"type\":\"long\",\"doc\":\"comment\"}," +
+        "      {\"id\":3,\"name\":\"z\",\"required\":true,\"type\":\"long\"}" +
+        "      ]" +
+        "    }" +
+        "  ]," +
+        "  \"partition-spec\":[]," +
+        "  \"default-spec-id\":0," +
+        "  \"partition-specs\":[{\"spec-id\":0,\"fields\":[]}]," +
+        "  \"last-partition-id\":999,\"default-sort-order-id\":1," +
+        "  \"sort-orders\":[" +
+        "    {\"order-id\":1,\"fields\":[" +
+        "        {\"transform\":\"identity\",\"source-id\":2,\"direction\":\"asc\",\"null-order\":\"nulls-first\"}," +
+        "        {\"transform\":\"bucket[4]\",\"source-id\":3,\"direction\":\"desc\",\"null-order\":\"nulls-last\"}" +
+        "      ]" +
+        "    }" +
+        "  ]," +
+        "  \"properties\":{\"owner\":\"hank\"}," +
+        "  \"current-snapshot-id\":-1," +
+        "  \"refs\":{},\"snapshots\":[],\"snapshot-log\":[],\"metadata-log\":[]},\"config\":{\"foo\":\"bar\"}}";
+
+    assertRoundTripSerializesEquallyFrom(fullJson.replaceAll("\\s", ""), respV1);
+
+    TableMetadata metaV2 = buildV2TableMetadataWithTagsAndSnapshotHistory();
+    LoadTableResponse respV2 = LoadTableResponse.builder()
+        .withTableMetadata(metaV2)
+        .addAllConfig(CONFIG)
+        .build();
+    Snapshot currentSnapshot = metaV2.snapshots().get(1);
+    Snapshot previousSnapshot = metaV2.snapshots().get(0);
+    String fullExampleInstanceJson = "{" +
+        "  \"metadataLocation\":\"s3://bucket/test/location/metadata/v1.metadata.json\"," +
+        "  \"metadata\":{" +
```

Comment:
Could you just use `TableMetadataParser` and build a string around it? There are similar tests in `TestCreateTableRequest` and that works great. I don't think it's very valuable to have complicated inline JSON like this.

Reply:
Updated to use `TableMetadataParser` as well as read in from existing external JSON files.